File: app/src/main/java/com/example/googleauthenticatorclient/NewAccountActivity.java
package com.example.googleauthenticatorclient;

import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import com.warrenstrange.googleauth.GoogleAuthenticator;
import com.warrenstrange.googleauth.GoogleAuthenticatorKey;

public class NewAccountActivity extends AppCompatActivity {


    EditText etInputCode;
    Button btnSubmit;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_new_account);
        System.setProperty("com.warrenstrange.googleauth.rng.algorithmProvider", "AndroidOpenSSL");
        final GoogleAuthenticator gAuth = new GoogleAuthenticator();

        final GoogleAuthenticatorKey key = gAuth.createCredentials();
        final String sharedKey = key.getKey();
        TextView tvKey = findViewById(R.id.key);
        tvKey.setText(key.getKey());

        etInputCode = findViewById(R.id.input_code);
        btnSubmit = findViewById(R.id.submit);

        btnSubmit.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                try {
                    Integer code = Integer.parseInt(etInputCode.getText().toString());
                    boolean isCodeValid = gAuth.authorize(sharedKey, code);
                    if(isCodeValid){
                        SharedPreferences sharedPreferences = getSharedPreferences("details", Context.MODE_PRIVATE);
                        sharedPreferences.edit().putString("sharedKey",sharedKey).apply();
                        startActivity(new Intent(NewAccountActivity.this,MainActivity.class));
                        Toast.makeText(NewAccountActivity.this,"Registered Successfully!",Toast.LENGTH_SHORT).show();
                    }else {
                        Toast.makeText(NewAccountActivity.this,"Incorrect Code!",Toast.LENGTH_SHORT).show();
                    }
                }catch (Exception e){
                    Toast.makeText(NewAccountActivity.this,"Enter VALID Code!",Toast.LENGTH_SHORT).show();
                }
            }
        });



    }
}

File: app/src/main/java/com/example/googleauthenticatorclient/ui/NewAccountActivity.kt
package com.example.googleauthenticatorclient.ui

import android.arch.lifecycle.ViewModelProviders
import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.View
import android.widget.Toast
import com.example.googleauthenticatorclient.R
import com.example.googleauthenticatorclient.data.Repository
import com.example.googleauthenticatorclient.di.DaggerHomeActivityComponent
import com.example.googleauthenticatorclient.di.RepositoryModule
import com.example.googleauthenticatorclient.ui.viewModel.NewAccountActivityViewModel
import com.example.googleauthenticatorclient.ui.viewModel.NewAccountActivityViewModelFactory
import kotlinx.android.synthetic.main.activity_new_account.*
import javax.inject.Inject

class NewAccountActivity : AppCompatActivity() {

    @Inject
    lateinit var repository: Repository

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_new_account)

        val component = DaggerHomeActivityComponent.builder()
                .repositoryModule(RepositoryModule(this))
                .build()
        component.init(this)

        val newAccountActivityViewModelFactory = NewAccountActivityViewModelFactory(repository)
        val viewModel = ViewModelProviders.of(this,newAccountActivityViewModelFactory ).get(NewAccountActivityViewModel::class.java)

        btn_submit.setOnClickListener { View.OnClickListener {
            val code = et_input_code.text.toString()
            try {
                if(viewModel.verifyCode(code.toInt())){
                    viewModel.setSharedKey(code)
                    startActivity(Intent(this@NewAccountActivity,HomeActivity::class.java))
                    Toast.makeText(this@NewAccountActivity, getString(R.string.register_successfully), Toast.LENGTH_SHORT).show()
                }else{
                    Toast.makeText(this@NewAccountActivity, getString(R.string.incorrect_code), Toast.LENGTH_SHORT).show()
                }
            } catch (e: Exception) {
                Toast.makeText(this@NewAccountActivity, getString(R.string.enter_valid_code), Toast.LENGTH_SHORT).show()
            }
        } }
    }
}
